pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionCommon.java

```diff
@@ -120,8 +121,20 @@ public static Class<?>[] getFunctionTypes(Class userClass, boolean isWindowConfi
         } else {
             if (Function.class.isAssignableFrom(userClass)) {
                 typeArgs = TypeResolver.resolveRawArguments(Function.class, userClass);
+                if (typeArgs[1].equals(Record.class)) {
```

Comment:
we need some unit test coverage for these branches

Reply:
This branch is covered by IT in PulsarFunctionTest.
I could cover also the j.u.f.Function if needed.
I don't think there are UT for the non `Record` types. Should we add ones ?

Reply:
I'll add UT for all the cases (Record, non-Record, j.u.f.Function, WindowFunction, ...)

Reply:
done